### ISSUE TITLE
Prefer-switch should ignore if-else statements with multiple OR conditions

### ISSUE THREAD

**Issue by joyt** [palantir/tslint#3719]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 5.3.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
export enum ItemType {
    FIRST = "FIRST",
    SECOND = "SECOND",
    THIRD = "THIRD",
}

if (
    item.type === ItemType.FIRST ||
    item.type === ItemType.SECOND ||
    item.type === ItemType.THIRD
) {
    ...
}
```

with `tslint.json` configuration:

```json
{
  "extends": ["tslint:latest", "tslint-react"],
  "rules": {
    "arrow-parens":  [true, "ban-single-arg-parens"],
    "await-promise": true,
    "binary-expression-operand-order": true,
    "encoding": true,
    "no-boolean-literal-compare": true,
    "no-console": [true, "log", "debug"],
    "no-default-export": true,
    "no-for-in-array": true,
    "no-inferrable-types": true,
    "no-import-side-effect": true,
    "no-invalid-this": true,
    "no-irregular-whitespace": true,
    "no-mergeable-namespace": true,
    "no-shadowed-variable": true,
    "no-unnecessary-callback-wrapper": true,
    "no-unnecessary-qualifier": true,
    "no-unnecessary-type-assertion": [true, "any"],
    "number-literal-format": true,
    "object-literal-key-quotes": [true, "as-needed"],
    "ordered-imports": [true, {
      "import-sources-order": "lowercase-last",
      "named-imports-order": "lowercase-last",
      "module-source-path": "full"
    }],
    "prefer-method-signature": true,
    "prefer-readonly": true,
    "return-undefined": true,
    "restrict-plus-operands": true,
    "prefer-switch": true,
    "triple-equals": [
      true,
      "allow-undefined-check",
      "allow-null-check"
    ],
    "variable-name": [true,
      "ban-keywords",
      "allow-leading-underscore",
      "allow-pascal-case"
    ],
    "max-classes-per-file": false,
    "no-object-literal-type-assertion": false,
    "no-string-literal": false,
    "object-literal-sort-keys": false,
    "member-ordering": false,
    "jsx-no-multiline-js": false,
    "forin": false
  }
}

```

#### Actual behavior
Getting a lint error for the "prefer-switch" switch at the `if` statement condition.

#### Expected behavior
No error, tslint should allow `if-else` statements over `switch` when you have one large OR  combination `===` check (especially if you don't allow fallthroughs)

**Comment by simonhaenisch** [comment]
A team member just brought this to my attention... I actually find the following reads a bit easier:

```ts
switch (item.type) {
  case ItemType.FIRST:
  case ItemType.SECOND:
  case ItemType.THIRD:
    // ...
    break;

  default:
     // this is the else branch
}
```

Or does this also happen for `if` statements that don't have an `else`? (In which case I would also see that as a bug.)

Disallowing empty fall-throughs would be a bit silly anyway, right? At least I haven't encountered any linter that disallows that.
